tests/python_package_test/test_engine.py

```diff
@@ -1072,6 +1072,46 @@ def test_cvbooster():
     assert ret < 0.15
 
 
+def test_cvbooster_save_load(tmp_path):
+    X, y = load_breast_cancer(return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    params = {
+        'objective': 'binary',
+        'metric': 'binary_logloss',
+        'verbose': -1,
+    }
+    nfold = 3
+    lgb_train = lgb.Dataset(X_train, y_train)
+
+    def predict(cv_booster):
+        return np.array(cv_booster.predict(X_test))
+
+    cv_res = lgb.cv(params, lgb_train,
+                    num_boost_round=10,
+                    nfold=nfold,
+                    callbacks=[lgb.early_stopping(stopping_rounds=5)],
+                    return_cvbooster=True)
+    cvbooster = cv_res['cvbooster']
+
+    preds = predict(cvbooster)
+
+    model_path_txt = str(tmp_path / 'lgb.model')
+    model_path_pkl = str(tmp_path / 'lgb.pkl')
+
+    cvbooster.save_model(model_path_txt)
+    with open(model_path_pkl, 'wb') as f:
```

Comment:
This project tries to support `pickle`, `joblib`, and `cloudpickle`.

Could you please modify this test to use all 3 of those serializers?

Some references to similar tests:

https://github.com/microsoft/LightGBM/blob/56ccea4243aba4c538ebca9de8c67ee0c9bc60eb/tests/python_package_test/test_callback.py#L33

https://github.com/microsoft/LightGBM/blob/56ccea4243aba4c538ebca9de8c67ee0c9bc60eb/tests/python_package_test/test_callback.py#L8-L22

Reply:
@jameslamb 
Thanks! I'll give it a try. Can I move `SERIALIZERS` and `pickle_and_unpickle_object` from `test_serializers.py` to `tests/python_package_test/utils.py` to reuse them? I think these do not appear to be callback-specific functions, but rather generic codes.

Reply:
Yep, moving those and re-using them sounds good to me!

Reply:
Thanks, done in  [3ec7fd0](https://github.com/microsoft/LightGBM/pull/5160/commits/3ec7fd02ebefa64b560135b14fcd3f44ab0affda). Now it seemed too many things to write everything in the single test, so I've extracted the pickle test into a separate function.